src/tome.pm

```diff
@@ -144,18 +146,19 @@ saved in the array previously and uses it as a return value.
 
 sub message($$) {
    my $self = shift;
-   my $txt = shift // die 'Insufficient arguments';
-
-   my $rndkey = (keys %tome)[rand keys %tome];
+   my $txt  = shift // die 'Insufficient arguments';
    
-   if ($txt =~ m{[^\s\n+\d-]}) {
+   my $rnd_idx = int rand scalar @tome; 
+   my $rnd_msg = $tome[$rnd_idx];
+   
+   if ($txt =~ m{[^\s+\d-]}) {
       my $txt = (split '\n', $txt)[0];
-
-      delete $tome{ $rndkey } if (keys %tome >= $tome_max);
-      $tome{$txt} = substr $txt, 0, $tome_msg_max;
+      $txt = substr ($txt, 0, $tome_msg_max) if length $txt >= $tome_msg_max;
```

Comment:
What for it that if:
```perl
if length $txt >= $tome_msg_max;
```
You know for sure that it is needed? May be you've performed some speed tests?